cli/src/cmd/forge/create.rs

```diff
@@ -232,20 +229,55 @@ impl CreateArgs {
         }
 
         let (deployed_contract, receipt) = deployer.send_with_receipt().await?;
+        let address = deployed_contract.address();
         if self.json {
             let output = json!({
                 "deployer": deployer_address,
-                "deployedTo": deployed_contract.address(),
+                "deployedTo": address,
                 "transactionHash": receipt.transaction_hash
             });
             println!("{output}");
         } else {
             println!("Deployer: {deployer_address:?}");
-            println!("Deployed to: {:?}", deployed_contract.address());
+            println!("Deployed to: {:?}", address);
             println!("Transaction hash: {:?}", receipt.transaction_hash);
+        };
+
+        if !self.verify {
+            return Ok(())
         }
 
-        Ok(())
+        println!("Starting contract verification...");
+        let constructor_args = if !args.is_empty() {
+            Some(String::from_utf8(
+                abi.constructor()
+                    .ok_or(eyre::eyre!("could not find constructor"))?
+                    .encode_input(bin.to_vec(), &args)?,
+            )?)
+        } else {
+            None
+        };
+        let num_of_optimizations =
+            if self.opts.compiler.optimize { self.opts.compiler.optimizer_runs } else { None };
+        let verify = verify::VerifyArgs {
+            address,
+            contract: self.contract,
+            compiler_version: None,
```

Comment:
~~why is the compiler_version missing here?~~~
ah because autodetected in next step, right?

Reply:
that's right, it's taken from artifacts anyway